src/hypothesis/extra/datetime.py

```diff
@@ -90,36 +222,60 @@ def datetimes(allow_naive=None, timezones=None, min_year=None, max_year=None):
     ]
     if allow_naive is None:
         allow_naive = not timezones
-    if not (timezones or allow_naive):
-        raise InvalidArgument(
-            u'Cannot create non-naive datetimes with no timezones allowed'
-        )
     return DatetimeStrategy(
         allow_naive=allow_naive, timezones=timezones,
         min_year=min_year, max_year=max_year,
+        min_datetime=min_datetime, max_datetime=max_datetime
     )
 
 
 @defines_strategy
-def dates(min_year=None, max_year=None):
-    """Return a strategy for generating dates."""
+def dates(min_year=None, max_year=None, min_date=None, max_date=None):
+    """Return a strategy for generating dates.
+
+    The arguments are as for :py:func:`datetimes`, including the deprecation
+    of ``min_year`` and ``max_year``.  If ``min_date`` or ``max_date`` are
+    datetime objects, any timezone information is discarded and they are
+    treated as dates.
+
+    .. code-block:: pycon
+
+        >>> from hypothesis.extra.datetime import dates
+        >>> dates().example()
+        datetime.date(1687, 3, 23)
+        >>> dates().example()
+        datetime.date(9565, 5, 2)
+
+    """
+    # This may give validation errors which refer to min_datetimes or
+    # max_datetimes, but the alternative is to duplicate or complicate
+    # validation considerably.
+    if isinstance(min_date, dt.datetime):
+        min_date = min_date.date()
+    if isinstance(max_date, dt.datetime):
+        max_date = max_date.date()
     return datetimes(
         allow_naive=True, timezones=[],
         min_year=min_year, max_year=max_year,
-    ).map(datetime_to_date)
-
-
-def datetime_to_date(dt):
-    return dt.date()
+        min_datetime=min_date, max_datetime=max_date
+    ).map(dt.datetime.date)
 
 
 @defines_strategy
 def times(allow_naive=None, timezones=None):
```

Comment:
Not yet.  Changing how I handle tz-aware datetime bounds will make it useful though, so I will provide a `min_time` and `max_time` argument.  (with no deprecation logic, yay!)